### DISCUSSION TITLE
[slog_scope] scope() and await

### DISCUSSION THREAD

**Discussion by Geobert** [slog-rs/slog#314] [Q&A] [answered]
Hi,

I'd like to add key:value to a logger and to do so, I'm calling `slog_scope::scope` where a pass a new logger with the needed key:values.
the issue is that the code in the function is async and the `await` are not accepted.

Is there a way to work around that?

**Comment by dpc** [comment]
If you're writting `async` code, I would consider using `tracing` ecosystem instead anyway.

**Comment by dpc** [comment, marked as answer]
Other than this, `slog-scope` would need a an `async` feature flag and a new set of functions that are async-enabled. PRs welcome, but probably easier to just use `tracing` and get other async goodies.
